### ISSUE TITLE
Implement an accessible dialog

### ISSUE THREAD

**Issue by HugoGiraudel** [morkro/happy-plants#46]
The current dialog implementation in use when creating a category on the fly when adding a new plant or the one to manage the modules is fundamentally inaccessible. My recommendation would be to create a thin Vue component wrapper a11y-dialog to avoid having to reimplement the underlying logic.

A few things that should happen (most of them done by a11y-dialog):

- [ ] The focus should be moved inside the dialog on open.
- [ ] The focus should be moved back where it was before opening.
- [ ] The focus should be trapped within the dialog.
- [ ] The <kbd>ESC</kbd> key should close the dialog.
- [ ] The main content of the page should be hidden from the accessibility tree.
- [ ] The dialog container should be present in the accessibility tree.
- [ ] The dialog should have `dialog` role in the accessibility tree.
- [ ] The dialog should have an associated name in the accessibility tree.

**Comment by morkro** [comment]
That's been done with `vue-a11y-dialog`.
